django/db/backends/base/creation.py

```diff
@@ -260,6 +260,8 @@ def destroy_test_db(self, old_database_name=None, verbosity=1, keepdb=False, suf
                 action,
                 self._get_database_display_str(verbosity, test_database_name),
             ))
+        if self.connection.vendor == 'oracle':
+            test_database_name = suffix
```

Comment:
To fix this, I changed ``_destroy_test_db```'s signature to use the cloned db's ```test_settings_dict``` instead of just the ```test_database_name```. Let me know your thoughts on that change. I wanted to add docs for it but I want to hear your opinion on it in case I should change it before finalizing the patch with the relevant documentation.